python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py

```diff
@@ -2710,6 +2713,23 @@ def aten〇leaky_relu〡dtype(self_rank_dtype: Tuple[int, int], negative_slope:
     dtypes = [self_dtype, negative_slope_dtype]
     return promote_dtypes(ranks, dtypes)
 
+@check_dtype_function(
+    _check_tensors_with_the_same_dtype(num_of_tensors=1, error_types={torch.bool}, alpha=1, scale=1, input_scale=2) +
+    _check_tensors_with_the_same_dtype(num_of_tensors=1, error_types={torch.bool, torch.int8, torch.uint8, torch.int16, torch.int32, torch.int64}, alpha=1.0, scale=1.0, input_scale=2.0))
+def aten〇elu〡dtype(self_rank_dtype: Tuple[int, int], alpha: Union[int, float, complex] = 1, scale: Union[int, float, complex] = 1, input_scale: Union[int, float, complex] = 1) -> int:
+    self_rank, self_dtype = self_rank_dtype
+    assert self_dtype != torch.bool
+    ranks: List[Optional[int]] = [self_rank, None]
+    alpha_dtype = get_dtype_of_scalar(alpha)
+    scale_dtype = get_dtype_of_scalar(scale)
+    input_scale_dtype = get_dtype_of_scalar(input_scale)
+
+    if any([is_float_dtype(d) for d in [alpha_dtype, scale_dtype, input_scale_dtype]]):
+        assert not is_integer_dtype(self_dtype)
+
+    dtypes = [self_dtype, input_scale_dtype]
+    return promote_dtypes(ranks, dtypes)
+
```

Comment:
Hi @123epsilon, I think the dtype computation for this op does not involve the scalar types. I tried different variants for this op, and it turned out that:
1.) The input tensor always has to have floating-point values(f16, f32, f64).
2.) Irrespective of the scalar types, the result type is the same as the input dtype.

Reply:
Ok, I've amended the dtype function - in the dtype tests however we still require checking whether the tensor is an int tensor in the case that the params are floating point - otherwise we do not match the behavior from [here](https://github.com/pytorch/pytorch/blob/256fed02e930210dbcd7e5e23fcf142362098c2a/torch/_refs/nn/functional/__init__.py#L231C34-L231C34) and the dtype tests fail, similarly with asserting that the tensor is not of bool type.